### PR TITLE
Migrating ca-form builder

### PR DESCRIPTION
**Pull request by norisalsaadie**
- components/ca-form-builder.css - direct clone

- components/ca-form-builder.js - migrated to ES6 + modules - reflection and API was removed.

- components/helpers/keyboard.js - migrated from helpers.js

- examples/ca-form-builder.html - example clone

- examples/data/ca-form-builder-schema.json - example and test data (forced the tests to use the example data)

- examples/data/ca-form-builder-toolbox.json - example and test data (forced the tests to use the example data)

- test/ca-big-red-button-test.html - removed debugger statement

- test/ca-form-builder-test.html - almost a 1:1 clone from the old, uses real HTTP requests as the previous one

### CODE REVIEW

File: components/ca-form-builder.js [hunk 1]
```diff
@@ -0,0 +1,1379 @@
...
+                // set the source element as the selected tool
+                this.setSelectedTool.call(this, src);
+            } else if (tagName === 'BUTTON') {
+                // determine which tool has been selected
+                const selectedTool = this.getSelectedTool.call(this);
+
+                // if there is a selected tool
+                if (selectedTool.tool) {
+                    // create a new schema by cloning the selected tool from the form builder grand schema
+                    const newSchema = cloneObject(this.schemaToolbox.find(s => s.id === selectedTool.schemaType));
```
Comment (norisalsaadie) [inline components/ca-form-builder.js]: The latter, but good point.

File: components/ca-form-builder.js [hunk 2]
```diff
@@ -0,0 +1,1379 @@
...
+        urnToJson() {
+            const result = {
+                namespace: null,
+                component: null,
+                tool: null,
+                fieldName: null,
+                sequence: null
+            };
+            const validKeys = 'namespace:component:tool:fieldName:sequence';
+            const parts = arguments[0].split(':'); // eslint-disable-line prefer-rest-params
```
Comment (norisalsaadie) [inline components/ca-form-builder.js]: I'll meet you halfway - spread params ...args

File: components/ca-form-builder.js [hunk 3]
```diff
@@ -0,0 +1,1379 @@
...
+                const previewFormSchema = this.previewForm.schema[0];
+
+                // get hold of the schema for the selected element
+                const props = previewFormSchema.properties[selectedElement.dataKey];
+
+                // if the properties exists in the schema of the preview form
+                if (props) {
+                    // generate the urn for this new schema (adding one to the sequence for the new element)
+                    const urn = this.urnToJson(props.id, {
+                        sequence: (props.sequence * 1) + 1
```
Comment (kevmarchant) [inline components/ca-form-builder.js]: hahaha!

File: components/ca-form-builder.css [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+    top: 10px;
+    min-height: 400px;
+    min-width: 600px;
+    display: block;
+}
+ca-form-builder ca-form {
+    overflow: auto;
+    display: block;
+}
+/* place an outline around the selected element */
```
Comment (kevmarchant) [inline components/ca-form-builder.css]: remove comment

File: components/ca-form-builder.css [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+}
+ca-form-builder .ca-form-builder-preview ca-form label[data-selected="true"]:after,
+ca-form-builder .ca-form-builder-preview ca-form label.required[data-selected="true"]:after {
+    color: red;
+    font-size: 14px;
+    top: 8px;
+    -webkit-transition: all .25s;
+    transition: all .25s;
+}
+/* style the container for the toolbox buttons */
```
Comment (kevmarchant) [inline components/ca-form-builder.css]: I don't mind comments when they explain something I can't ascertain for myself in 0.3 seconds!

File: components/ca-form-builder.css [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+
+.ca-form-builder-toolbox-buttons {
+    padding: 10px 10px 6px 10px;
+    border: 1px solid #bbb;
+    border-radius: 3px;
+    background-color: #f8f8f8;
+    margin-bottom: 4px;
+}
+/* style each tool in the tool box */
```
Comment (kevmarchant) [inline components/ca-form-builder.css]: really?

File: components/ca-form-builder.js [hunk 7]
```diff
@@ -0,0 +1,1377 @@
...
+                // determine which tool has been selected
+                const selectedTool = this.getSelectedTool.call(this);
+
+                // if there is a selected tool
+                if (selectedTool.tool) {
+                    // create a new schema by cloning the selected tool from the form builder grand schema
+                    const newSchema = cloneObject(this.schemaToolbox.find(s => s.id === selectedTool.schemaType));
+
+                    // need to show the property form in the property area
+                    this.propertyFormShow.call(this);
```
Comment (kevmarchant) [inline components/ca-form-builder.js]: remove the .call()s

File: components/ca-form-builder.js [hunk 8]
```diff
@@ -0,0 +1,1377 @@
...
+                                const thisElement = this.getThisElement.call(this, selectedElement.dataKey);
+
+                                // try to click on this element but if we couldn't
+                                if (this.clickThisElement.call(this, thisElement.dataKey) === false) {
+                                    // click the first element on the page
+                                    this.clickFirstElement.call(this);
+                                }
+                            }
+                        }
+                        // return false;
```
Comment (kevmarchant) [inline components/ca-form-builder.js]: remove comment

**Pull request merged**